### ISSUE TITLE
Bedrock players are able to go through doors even if the event cancelled.

### ISSUE THREAD

**Issue by TraceLosu** [GeyserMC/Geyser#2819]
### Describe the bug

Bedrock players can go through doors or trapdoors, even if a plugin has canceled the event. Plugins like GriefPrevention, and LWCX have broken door protection because of it. 

### To Reproduce

1. Run a Paper server with GeyserMC, and any form of protection plugin (what I used was LWCX and GriefDefender).
2. Build an enclosure with only a single door as it's entrance.
3. Lock/Claim said entrance using a protection plugin.
4. Using a bedrock account, stand in front of the door and repeatedly open the door while holding the forward button.
5. For Java players, the door does not open at all. But, the player goes through the door.

### Expected behaviour

The player should not be able to go through the door.

### Screenshots / Videos

[Video](https://cdn.discordapp.com/attachments/613168464634576897/940477712718499850/Jdk17.0.1_12_2022.02.08_-_12.09.02.03.mp4)

### Server Version and Plugins

- Geyser; `This server is running Geyser version 2.0.0-SNAPSHOT (git-master-9d908c5) (Java: 1.18 - 1.18.1, Bedrock: 1.17.30 - 1.18.0/1.18.1/1.18.2)`
- Paper: `This server is running Paper version git-Paper-187 (MC: 1.18.1) (Implementing API version 1.18.1-R0.1-SNAPSHOT) (Git: 0cc2503)`

### Geyser Dump

https://dump.geysermc.org/m0xiMh13GOr1AqPUqeRYEnjkLQky8Bg5

### Geyser Version

2.0.0-SNAPSHOT (git-master-9d908c5)

### Minecraft: Bedrock Edition Version

1.18.2

**Comment by hausemaster3734** [comment]
This shouldn't happen as the door opening event is cancelled.

**Comment by Wolvend** [comment]
Having same issue. No fix currently.

**Comment by tdherring** [comment]
I have created a plugin which addresses this issue by tracking Bedrock player positions and teleporting those who attempt to access "locked" doors/trapdoors to their original location. This should work for *all* plugins experiencing the issue since it listens for the event cancellation, which Geyser is seemingly ignoring. Your feedback is appreciated!

Please see: [Geyser Interaction Fix](https://github.com/tdherring/geyser-interaction-fix)

**Comment by Camotoy** [comment]
Looks nice. I would recommend using the Geyser or Floodgate API so that linked players will also cooperate. Geyser doesn't really use the Bukkit API so I'm not sure if this fits in Geyser mainline. Though, I'm also not exactly sure why Bedrock players are affected and I need to test.

**Comment by tdherring** [comment]
> Looks nice. I would recommend using the Geyser or Floodgate API so that linked players will also cooperate. Geyser doesn't really use the Bukkit API so I'm not sure if this fits in Geyser mainline. Though, I'm also not exactly sure why Bedrock players are affected and I need to test.

Good point, didn't think of that edge case - I'll work on moving over to the Geyser/Floodgate API now! It's an odd issue definitely, one thing I can say is that doors on Bedrock seem to ping back to shut a lot slower than Java. As for the root cause, no idea - maybe the translation overhead? I doubt it though.

**Comment by Z3r0xh** [comment]
I just tested this plugin and it doesn't work anymore in recent versions. Users can still open doors and trapdoors

**Comment by stooneg0mes** [comment]
Hello! Do we have any news on this issue?

**Comment by JosTheDude** [comment]
Any update? This bug is still present.
